File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaUniversidadBO/BibliotecaBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;

namespace SoftBibliotecaPublicacionBO
{
    public class BibliotecaBO : BaseBO
    {
        public int insertar(string nombre, string ubicacion, int idUniversidadAsociada)
        {
            return this.WsCliente.biblioteca_insertar(nombre, ubicacion, idUniversidadAsociada);
        }

        public int modificar(int idBiblioteca, string nombre, string ubicacion,
                                 int idUniversidadAsociada)
        {
            return this.WsCliente.biblioteca_modificar(idBiblioteca, nombre, ubicacion, idUniversidadAsociada);
        }

        public int eliminar(int idBiblioteca)
        {
            return this.WsCliente.biblioteca_eliminar(idBiblioteca);
        }

        public BindingList<biblioteca> listarTodos(int idUniversidadAsociada, int limite)
        {
            try
            {
                biblioteca[] bibliotecas = this.WsCliente.biblioteca_listarTodos(idUniversidadAsociada, limite);
                if (bibliotecas == null) return new BindingList<biblioteca>();
                return new BindingList<biblioteca>(bibliotecas);
            }
            catch (Exception ex)
            {
                return new BindingList<biblioteca>();
            }
        }

        public biblioteca obtenerPorId(int idBiblioteca)
        {
            return this.WsCliente.biblioteca_obtenerPorId(idBiblioteca);
        }

        public BindingList<biblioteca> mostrarBibliotecasDeUniversidadesDelProgramador(int idUsuario, int[] idUniversidadesAsociadasDelBibliotecario)
        {
            try
            {
                biblioteca[] bibliotecas = this.WsCliente.mostrarBibliotecasDeUniversidadesDelProgramador(idUsuario, idUniversidadesAsociadasDelBibliotecario);
                if (bibliotecas == null) return new BindingList<biblioteca>();
                return new BindingList<biblioteca>(bibliotecas);
            }
            catch (Exception ex)
            {
                return new BindingList<biblioteca>();
            }
        }
    }
}

File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/BibliotecaBO.java
package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.localidad.model.Biblioteca;

public interface BibliotecaBO extends Remote{
    
    public Integer insertar(String nombre,String ubicacion,Integer idUniversidadAsociada) throws RemoteException;

    public Integer modificar(Integer idBiblioteca,String nombre,String ubicacion,
                             Integer idUniversidadAsociada) throws RemoteException;

    public Integer eliminar(Integer idBiblioteca) throws RemoteException;

    public ArrayList<Biblioteca> listarTodos(Integer idUniversidadAsociada, Integer limite) throws RemoteException;

    public Biblioteca obtenerPorId(Integer idBiblioteca) throws RemoteException;
    
    public ArrayList<Biblioteca> mostrarBibliotecasDeUniversidadesDelProgramador(Integer idUsuario, 
            ArrayList<Integer>idUniversidadesAsociadasDelBibliotecario) throws RemoteException;
    
}
